### DISCUSSION TITLE
How to reject empty string argument?

### DISCUSSION THREAD

**Discussion by hyochan** [prisma/prisma#3109] [Q&A] [answered]
Followed by [nexus](https://nexus.js.org/docs/api-args) document, I'd like to prevent falsy argument (in javascript) which also includes empty string.

Below query doesn't throw any error if I try it.
```
mutation signInEmail {
  signInEmail(
    email: ""
    password: ""
  ) {
    token
    user {
      id
    }
  }
}
```

I've tried `required` and `nullable`. Is this really expected?
I am afraid to prevent these falsy arguments manually in server-side.

```
if (!email || !password) throw new Error();
```

I have to do above to all other resolvers. Is there any better way around?

**Comment by ryands17** [comment, marked as answer]
Hey @hyochan 👋 
Required and nullable only check for `null` and not falsy values. So an empty string is valid in that case.
Could you create a feature request [here](https://github.com/graphql-nexus/nexus/issues/new?assignees=&labels=type%2Ffeat&template=10-feature.md&title=) so that they can look into adding custom validations.

Also you can check out [this](https://github.com/huv1k/nexus-arguments-validation) validation package as it has custom validations.

**Reply by hyochan** [reply]
Thanks for the response. I've just created a feature request in [#1278](https://github.com/graphql-nexus/nexus/issues/1278).

The link you've shared doesn't seem to be maintained since 4months ago.

Still in `0.0.0-preview1` so I am afraid of using it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
